fix(backup)_: use pubkey suffix instead of prefix

jrainville opened the pull request:
Fixes https://github.com/status-im/status-desktop/issues/18731

Turns out our pubkeys all start with the same prefix, and I chose exactly the length of the prefix in the previous version, so all files had the same name. Using the suffix solves this

jrainville committed:
fix(backup)_: use profile keypair for backup files

Fixes https://github.com/status-im/status-desktop/issues/18731

Turns out our pubkeys all start with the same prefix, and I chose exactly the length of the prefix in the previous version, so all files had the same name.
Using the profile keypair solves this

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (25)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 3e46ca6d | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6874/1/) | 2025-08-26 20:52:40 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250826-205023-3e46ca-pr6874.aar) |
| :heavy_check_mark: | 3e46ca6d | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6874/1/) | 2025-08-26 20:53:09 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250826-205024-3e46ca-pr6874.zip) |
| :heavy_check_mark: | 3e46ca6d | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6874/1/) | 2025-08-26 20:53:43 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250826-205030-3e46ca-pr6874.zip) |
| :heavy_check_mark: | 3e46ca6d | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6874/1/) | 2025-08-26 20:55:33 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250826-205029-3e46ca-pr6874.zip) |
| :heavy_check_mark: | 3e46ca6d | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6874/1/) | 2025-08-26 20:57:06 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250826-205036-3e46ca-pr6874.zip) |
| :heavy_check_mark: | 3e46ca6d | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/1/) | 2025-08-26 20:58:10 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/1//consoleText) |
| :heavy_check_mark: | 3e46ca6d | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6874/1/) | 2025-08-26 21:00:32 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250826-205026-3e46ca-pr6874.zip) |
| :heavy_check_mark: | 3e46ca6d | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/1/) | 2025-08-26 21:18:00 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 1878dc3c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6874/2/) | 2025-08-26 21:01:50 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250826-205925-1878dc-pr6874.aar) |
| :heavy_check_mark: | 1878dc3c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6874/2/) | 2025-08-26 21:02:14 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250826-205926-1878dc-pr6874.zip) |
| :heavy_check_mark: | 1878dc3c | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6874/2/) | 2025-08-26 21:04:24 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250826-205931-1878dc-pr6874.zip) |
| :heavy_check_mark: | 1878dc3c | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6874/2/) | 2025-08-26 21:04:48 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250826-205930-1878dc-pr6874.zip) |
| :heavy_check_mark: | 1878dc3c | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6874/2/) | 2025-08-26 21:07:03 | ~7 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250826-205937-1878dc-pr6874.zip) |
| :heavy_check_mark: | 1878dc3c | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/2/) | 2025-08-26 21:07:11 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/2//consoleText) |
| :heavy_check_mark: | 1878dc3c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6874/2/) | 2025-08-26 21:09:32 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250826-205928-1878dc-pr6874.zip) |
| :heavy_check_mark: | 1878dc3c | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/2/) | 2025-08-26 21:43:10 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 21d31f33 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6874/3/) | 2025-08-27 14:37:52 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250827-143501-21d31f-pr6874.aar) |
| :heavy_check_mark: | 21d31f33 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6874/3/) | 2025-08-27 14:38:11 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250827-143502-21d31f-pr6874.zip) |
| :heavy_multiplication_x: | 21d31f33 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/3/) | 2025-08-27 14:39:51 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/3//consoleText) |
| :heavy_check_mark: | 21d31f33 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6874/3/) | 2025-08-27 14:39:58 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250827-143508-21d31f-pr6874.zip) |
| :heavy_check_mark: | 21d31f33 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6874/3/) | 2025-08-27 14:40:17 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250827-143507-21d31f-pr6874.zip) |
| :heavy_check_mark: | 21d31f33 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6874/3/) | 2025-08-27 14:41:43 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250827-143515-21d31f-pr6874.zip) |
| :heavy_check_mark: | 21d31f33 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6874/3/) | 2025-08-27 14:44:43 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250827-143504-21d31f-pr6874.zip) |
| :heavy_check_mark: | 21d31f33 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/3/) | 2025-08-27 15:00:31 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/3//consoleText) |
| :heavy_multiplication_x: | 21d31f33 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/4/) | 2025-08-27 16:47:25 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/4//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a8fde727 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6874/4/) | 2025-08-27 17:16:27 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250827-171322-a8fde7-pr6874.aar) |
| :heavy_check_mark: | a8fde727 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6874/4/) | 2025-08-27 17:16:48 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250827-171323-a8fde7-pr6874.zip) |
| :heavy_check_mark: | a8fde727 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6874/4/) | 2025-08-27 17:19:35 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250827-171328-a8fde7-pr6874.zip) |
| :heavy_check_mark: | a8fde727 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6874/4/) | 2025-08-27 17:19:35 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250827-171329-a8fde7-pr6874.zip) |
| :heavy_check_mark: | a8fde727 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/5/) | 2025-08-27 17:20:39 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/5//consoleText) |
| :heavy_check_mark: | a8fde727 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6874/4/) | 2025-08-27 17:21:02 | ~7 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250827-171335-a8fde7-pr6874.zip) |
| :heavy_check_mark: | a8fde727 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6874/4/) | 2025-08-27 17:25:42 | ~12 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250827-171325-a8fde7-pr6874.zip) |
| :heavy_check_mark: | a8fde727 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/4/) | 2025-08-27 17:38:27 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a0238661 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/android/job/PR-6874/5/) | 2025-08-27 17:19:08 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250827-171634-a02386-pr6874.aar) |
| :heavy_check_mark: | a0238661 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6874/5/) | 2025-08-27 17:20:09 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250827-171653-a02386-pr6874.zip) |
| :heavy_check_mark: | a0238661 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6874/5/) | 2025-08-27 17:24:04 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250827-171407-a02386-pr6874.zip) |
| :heavy_check_mark: | a0238661 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6874/5/) | 2025-08-27 17:24:50 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250827-171943-a02386-pr6874.zip) |
| :heavy_check_mark: | a0238661 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/ios/job/PR-6874/5/) | 2025-08-27 17:25:17 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250827-171947-a02386-pr6874.zip) |
| :heavy_check_mark: | a0238661 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/6/) | 2025-08-27 17:27:58 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6874/6//consoleText) |
| :heavy_check_mark: | a0238661 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6874/5/) | 2025-08-27 17:28:09 | ~7 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250827-172105-a02386-pr6874.zip) |
| :heavy_check_mark: | a0238661 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/5/) | 2025-08-27 18:04:21 | ~25 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6874/5//consoleText) |

saledjenic commented on node/get_status_node.go:
```diff
@@ -220,7 +220,7 @@ func (n *StatusNode) StartLocalBackup() error {
 			backupDir = filepath.Join(n.config.RootDataDir, "backups")
 		}
-		fullPath := filepath.Join(backupDir, fmt.Sprintf("%x_user_data.bkp", accountIdentifier[:4]))
+		fullPath := filepath.Join(backupDir, fmt.Sprintf("%x_user_data.bkp", accountIdentifier[len(accountIdentifier)-4:]))
```
In my opinion, the best identifier to use here is the profile key-uid. That aligns with other files/locations we store for the logged-in profile.

You can easily expose `func (db *Database) GetProfileKeypair() (*Keypair, error)` via `accounts` (or `wallet`) service like:
```
func (s *Service) GetProfileKeypair() (*accounts.Keypair, error) {

	return s.db.GetProfileKeypair()
}
``` 
and use it here:
```suggestion
		profileKeypair, err := n.accountsSrvc.GetProfileKeypair()
		if err != nil {
			return "", err
		}
		fullPath := filepath.Join(backupDir, fmt.Sprintf("%x_user_data.bkp", profileKeypair.KeyUID()))
```

jrainville replied:
@saledjenic done. I also added a second commit because I remembered that using the data dir on mobile doesn't work

jrainville committed:
feat(backup)_: set backup path to user's config folder if not set

This should make it work on mobile as well plus it sets a default to the setting

jrainville committed:
docs(tests)_: add instructions on how to run a single test
